From cba06623813e62f3e528e4b14fd7779ccc52a954 Mon Sep 17 00:00:00 2001
From: Nathan Muggli <nmuggli@google.com>
Date: Fri, 12 Apr 2024 13:46:47 -0600
Subject: [PATCH] Remove some unsupported options

After the star 3.14.0 uprev, there are a few options used in the code
that are not defined in the tsp143gt.ppd file.  This causes an index of
-1 being used to access an element in an array.  Check to make sure
there is a valid index before accessing these specific PPD options.
---
 src/rastertostar.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/rastertostar.c b/src/rastertostar.c
index 81fb31f..0eb2770 100755
--- a/src/rastertostar.c
+++ b/src/rastertostar.c
@@ -1165,7 +1165,9 @@ void jobSetup(struct settings_ settings, char *argv[])
     if (settings.modelNumber == TSP651 || settings.modelNumber == TSP654 || settings.modelNumber == HSP7000R || 
         settings.modelNumber == FVP10  || settings.modelNumber == TSP143 ||settings.modelNumber == TSP828L )
     {
+      if (settings.printDensity >= 0) {
         outputCommand(printDensityCommand[settings.printDensity]);
+      }
     }
 
 
@@ -1335,7 +1337,9 @@ void jobSetup(struct settings_ settings, char *argv[])
     
     if (settings.modelNumber == TSP143)
     {
+      if (settings.takenSensorHoldPrint >= 0) {
         outputCommand(takenSensorControlCommand[settings.takenSensorHoldPrint]);
+      }
     }
 
     outputCommand(rasterModeStartCommand);
-- 
2.44.0.683.g7961c838ac-goog

